src/composables/useBadgeContentfulData.js

```diff
@@ -0,0 +1,47 @@
+import { defaultBadges } from '#src/util/achievementUtils';
+import { gql } from 'graphql-tag';
+import logReadQueryError from '#src/util/logReadQueryError';
+
+export const badgeQuery = gql`query contentfulBadgeImage ($badgeKey: String!) {
+	contentful {
+		entries(contentKey: $badgeKey, contentType: "challenge")
+	}
+}`;
+
+export default function useBadgeContentfulData(client, route) {
```

Comment:
Ideally this composable gets unit testing, but it can be a follow-up task. Please log a ticket if it isn't done now. A great benefit of composables is that it is a great pattern for allowing testing in the FE, much like utils.